builder/amazon/common/step_pre_validate.go

```diff
@@ -76,13 +78,34 @@ func (s *StepPreValidate) Run(ctx context.Context, state multistep.StateBag) mul
 		ui.Say("Force Deregister flag found, skipping prevalidating AMI Name")
 		return multistep.ActionContinue
 	}
+
 	if s.AMISkipBuildRegion {
 		ui.Say("skip_build_region was set; not prevalidating AMI name")
 		return multistep.ActionContinue
 	}
 
 	ec2conn := state.Get("ec2").(*ec2.EC2)
 
+	// Validate VPC settings for non-default VPCs
```

Comment:
You asked offline if we should find a way to add acceptance tests for this; to start with, it'd probably be useful to at least add some unit tests. If you look at step_ami_region_copy_test.go you'll see that we've already created a mock ec2 connection that you can put into a statebag before running a step to test it out. In your case, since it turns out step_pre_validate has no unit tests so far, I'd probably simplify things by extracting your VPC validation into its own function on the step and run it independently to check that validation in that one area works as expected. Happy to hop on a call and walk through this if you want.

Reply:
Thanks for the guidance here. I'll refactor into a separate function and reach out offline to set up a pairing session